hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/NewOffenderEndpointTestService.kt

```diff
@@ -0,0 +1,49 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.indexer.services
+
+import com.microsoft.applicationinsights.TelemetryClient
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.prisonersearch.common.config.trackEvent
+import uk.gov.justice.digital.hmpps.prisonersearch.indexer.services.dto.nomis.OffenderBooking
+
+/**
+ * This service tests the new offender endpoint in prison-api by launching a coroutine to call the new endpoint and
+ * compare the results to the offender booking retrieved by the old endpoint, publishing the results to App Insights.
+ */
+@ConditionalOnProperty(value = ["diff.testNewOffenderEndpoint"], havingValue = "true")
+@Component
+class NewOffenderEndpointTestService(
+  private val telemetryClient: TelemetryClient,
+  private val nomisService: NomisService,
+) {
+  fun launchOffenderEndpointDiffTest(oldEndpointBooking: OffenderBooking) {
+    CoroutineScope(SupervisorJob()).launch {
+      trackDifferences(oldEndpointBooking)
+    }
```

Comment:
We run the call to test the new endpoint in an independent coroutine - if this coroutine fails for whatever reason it does not affect the main (parent) thread.